openid-4-verifiable-presentations-1_0.md

```diff
@@ -441,7 +441,7 @@ This specification defines the following values for the `client_id_scheme` param
 
 * `pre-registered`: This value represents the [@!RFC6749] default behavior, i.e., the Client Identifier needs to be known to the Wallet in advance of the Authorization Request. The Verifier metadata is obtained using [@!RFC7591] or through out-of-band mechanisms.
 
-* `redirect_uri`: This value indicates that the Verifier's redirect URI is also the value of the Client Identifier. In this case, the Authorization Request MUST NOT be signed, the Verifier MAY omit the `redirect_uri` Authorization Request parameter, and all Verifier metadata parameters MUST be passed using the `client_metadata` or `client_metadata_uri` parameter defined in (#vp_token_request). 
+* `redirect_uri`: This value indicates that the Verifier's Redirect URI (or Response URI when Response Mode `direct_post` is used) is also the value of the Client Identifier. The Authorization Request MUST NOT be signed. The Verifier MAY omit the `redirect_uri` Authorization Request parameter. All Verifier metadata parameters MUST be passed using the `client_metadata` or `client_metadata_uri` parameter defined in (#vp_token_request). 
```

Comment:
```suggestion
* `redirect_uri`: This value indicates that the Verifier's Redirect URI (or Response URI when Response Mode `direct_post` is used) is also the value of the Client Identifier. The Authorization Request MUST NOT be signed. The Verifier MAY omit the `redirect_uri` Authorization Request parameter. Verifier metadata parameters can be passed using the `client_metadata` or `client_metadata_uri` parameter defined in (#vp_token_request). 
```

I would not mandate this mechanism, as there could be more advanced methods for providing/fetching verifier's metadata, such as using verifiable attestations or OpenID Federation. Using MUST could restrict the specification by making implementations non-compliant if they adopt alternative approaches.

Reply:
I agree with Giuseppe's proposed change.

Reply:
But this is just in the context of the `redirect_uri` client_id_scheme - wouldn't "more advanced methods for providing/fetching verifier's metadata" be defined under a different client_id_scheme? 

Even so, the use of "All" to start that sentence could be read as meaning more than I think it's supposed to mean. Here's a slightly different suggestion:

```suggestion
* `redirect_uri`: This value indicates that the Verifier's Redirect URI (or Response URI when Response Mode `direct_post` is used) is also the value of the Client Identifier. The Authorization Request MUST NOT be signed. The Verifier MAY omit the `redirect_uri` Authorization Request parameter. Verifier metadata parameters MUST are passed using the `client_metadata` or `client_metadata_uri` parameter defined in (#vp_token_request). 
```

Reply:
I admit that I never understood why `The Authorization Request MUST NOT be signed`

Reply:
@peppelinux @selfissued Brian is right. this is only the text for client_id == redirect_uri case, so oid federation and verifier attestation cannot apply. are you ok with Brian's suggestion?

Reply:
To avoid confusion, I'd suggest to pre-prepend and additional remark, like 

`When `redirect_uri` is used` before ` the Verifier metadata parameters are passed using the `client_metadata` or `client_metadata_uri` parameter defined in (#vp_token_request). `

Reply:
but whether client_metadata parameters are used or not does not depend on redirect_uri but on the client_id_scheme. per PR #165

Reply:
@peppelinux re: appending "When redirect_uri is used"

This text is inside a bullet point that's solely for the redirect_uri client scheme. I don't think we need to clarify that we're talking about the redirect uri client scheme, and as Kristina says doing say may actually introduce new confusion.

I'm good with Brian's suggested text, when combined with Oliver's clarification.

Reply:
in this text we're saying the the client_metadata MUST be passed within the request, while this is not true since there might be several ases where the RP (client) metadata MUST/SHOULD be passed through other method, like thematic trusted lists, openid federation, metadata published at .well-known endpoint and so on depending by the trust framework used.

the most relevant evidence about this is the fact that we cannot pretend that the RP's metadata would not be attested by a trusted third party. This is up to the trust framework used, definitively out of the scopes of the openid4vp specs but necessarly implementable in it.

Reply:
I'm struggling to follow. This text is only applicable when `client_id_scheme` is `redirect_uri`, which it's fairly likely no one will use in production, and explicitly means "OpenID Federation is not being used" (because in that case `client_id_scheme` would be `entity_id`).

It is correct to say that the metadata must be in the request for this case. And once #17 is resolved the metadata will be clearly limited to be a pretty small set of metadata.